### PR TITLE
Add support for package source attribute

### PR DESCRIPTION
**Pull request by prolixalias**
- source param on package type
- optional on install subclass
#### Pull Request (PR) description
Allow source param on package

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

**Comment by dhoppe** [comment]
@prolixalias Could you please add an RSpec test that uses the new parameter?

File: manifests/install.pp [hunk 1]
```diff
@@ -8,6 +8,7 @@
   $manage_ruby_dependency,
   $puppet_master = true,
   $is_pe_server = $r10k::params::is_pe_server,
+  Optional[String[1]] $gem_source = undef,
```
Comment (bastelfreak) [inline manifests/install.pp:11]: can you please document the new parameter?

Comment (prolixalias) [reply manifests/install.pp:11]: Sure thing, thanks. All set...

**Comment by prolixalias** [comment]
@bastelfreak & @dhoppe - any interest in converting to PDK? Once this PR is through, I'd be happy to convert/adjust/PR... Otherwise, I'll go back about my regularly scheduled programming.

**Comment by bastelfreak** [comment]
@prolixalias thanks for the fixes. At the moment we use modulesync (https://github.com/voxpupuli/modulesync and https://github.com/voxpupuli/modulesync_config) to manage our modules. We haven't decided yet if we migrate to pdk or not. If we do, we would need to migrate all of our modules.

**Pull request merged**